org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/KotlinDefaultArgumentsFilter.java

```diff
@@ -48,24 +48,38 @@ static boolean isDefaultArgumentsMethodName(final String methodName) {
 		return methodName.endsWith("$default");
 	}
 
+	static boolean isDefaultArgumentsConstructor(final MethodNode methodNode) {
+		if (!methodNode.name.equals("<init>")) {
+			return false;
+		}
+		final Type[] argumentTypes = Type.getMethodType(methodNode.desc)
+				.getArgumentTypes();
+		if (argumentTypes.length < 2) {
+			return false;
+		}
+		return "kotlin.jvm.internal.DefaultConstructorMarker"
+				.equals(argumentTypes[argumentTypes.length - 1].getClassName());
+	}
+
 	public void filter(final MethodNode methodNode,
 			final IFilterContext context, final IFilterOutput output) {
 		if ((methodNode.access & Opcodes.ACC_SYNTHETIC) == 0) {
 			return;
 		}
-		if (!isDefaultArgumentsMethodName(methodNode.name)) {
-			return;
-		}
 		if (!KotlinGeneratedFilter.isKotlinClass(context)) {
 			return;
 		}
 
-		new Matcher().match(methodNode, output);
+		if (isDefaultArgumentsMethodName(methodNode.name)) {
```

Comment:
For consistency with the new `isDefaultArgumentsConstructor()` method I would rename this method to `isDefaultArgumentsMethod()`. Also passing a `MethodNode` both all callers do `methodNode.name` anyways.

Reply:
Done